install/kubernetes/cilium/values.yaml.tmpl

```diff
@@ -0,0 +1,1340 @@
+{{- $ciliumVersion := .version -}}
```

Comment:
values.yaml is the `actual` values for (helm) template, so with this change, we will have template for value for template :exploding_head:. 

While I understand the benefit of reducing scripting to `find and replace`, it might be confusing for maintenance. What do you think?

Reply:
@sayboras yeah template for values of another template. personally i find this a bit less error-prone than using `sed`, but maybe it's getting too meta 😅 i think @joestringer recently dealt with this Makefile, so would be good to get his take on this.

Reply:
@michi-covalent oh right, I just take a look at recent related PR, there was a few hurdles :sweat: :cry: . Personally, I agreed that scripting is a little bit imperative and error-prone (like you mentioned), but if it didn't change so often, we can still live with it. One example I can think of is Makefile, most of big projects have a giant Makefile. 

__Note__: In case that we decide to go with templating/generating files, how about using cuelang ? Cuelang is used for generating a bunch of connectivity check if I am not wrong :wink:

Reply:
@sayboras yeah that makes sense, didn't realize we already use `cuelang`. i just used the first one that came up on google 😄

Reply:
yeah I do wonder if we're just making things more complicated for contributors. Though that said, we already require editing the `values.yaml` and then running `make -C install/kubernetes` to make sure that the quick/experimental install YAMLs are generated. This would just change it to editing the `values.yaml.tmpl`.

I think if we're seriously worried about the robustness of the sed expression locating the `repository` lines in the `values.yaml` and then going to the next line and then replacing the tag, then it might be worth exploring this further? But either way it's very few folks who would have to deal with any trouble there: specifically, those who prepare new releases. So it's probably a bit more complicated than it needs to be.